pkg/config/shared.go

```diff
@@ -402,6 +396,49 @@ func (c InteropClient) Fetcher(ctx context.Context, blobConf BlobConfig) (fetch.
 	}
 }
 
+// InteropServer represents the server-side interoperability through LoRaWAN Backend Interfaces configuration.
+type InteropServer struct {
+	ConfigSource string         `name:"config-source" description:"Source of the interoperability server configuration (directory, url, blob)"`
+	Directory    string         `name:"directory" description:"OS filesystem directory, which contains interoperability server configuration"`
+	URL          string         `name:"url" description:"URL, which contains interoperability server configuration"`
+	Blob         BlobPathConfig `name:"blob"`
+
+	ListenTLS       string            `name:"listen-tls" description:"Address for the interop server to listen on"`
+	SenderClientCAs map[string]string `name:"sender-client-cas" description:"Path to PEM encoded file with client CAs of sender IDs to trust"`
+
+	BlobConfig BlobConfig `name:"-"`
+}
+
+// Fetcher returns fetch.Interface defined by conf.
+// If no configuration source is set, this method returns nil, nil.
+func (c InteropServer) Fetcher(ctx context.Context, blobConf BlobConfig) (fetch.Interface, error) {
+	// TODO: Remove detection mechanism (https://github.com/TheThingsNetwork/lorawan-stack/issues/1450)
+	if c.ConfigSource == "" {
+		switch {
+		case c.Directory != "":
```

Comment:
I don't think this is entirely backward compatible; if we have no `ConfigSource` but we have `len(c.SenderClientCAs) > 0`, we should revert to existing behavior and use `directory` config source with directory being empty or working dir

Reply:
Indeed, I copy-pasted from client implementation and did not account for this

Reply:
I don't think we need to be backwards compatible here. This code works on the newly added `sender-client-ca` option. Since it didn't exist yet in the past, there is no need for compatibility, right? And I don't think we need to be compatible with how other (similar) configs do this. 

Fine to leave this as it is, just saying I don't think it's necessary.